### PR TITLE
[TACHYON-1086] Make YARN AppMaster read from site-properties

### PR DESCRIPTION
**Pull request by apc999**
https://tachyon.atlassian.net/browse/TACHYON-1086

This PR improves the YARN integration so that our YARN setup scripts can read site-properties to decide how large memory to request and format on each worker container.

### CODE REVIEW

File: integration/yarn/src/main/java/tachyon/yarn/Client.java [hunk 1]
```diff
@@ -80,7 +80,7 @@
  */
 public final class Client {
   /** Main class to invoke ApplicationMaster. */
-  private final String mAmMainClass = ApplicationMaster.class.getName();
+  private final String AM_MAIN_CLASS = ApplicationMaster.class.getName();
```
Comment (calvinjia) [inline integration/yarn/src/main/java/tachyon/yarn/Client.java]: All caps is for `static final`?

File: integration/yarn/src/main/java/tachyon/yarn/Client.java [hunk 2]
```diff
@@ -297,13 +297,16 @@ private void setupAppMasterJar(LocalResource appMasterJar) throws IOException {
...
-          c.trim(), ApplicationConstants.CLASS_PATH_SEPARATOR);
+      Apps.addToEnvironment(appMasterEnv, classpath, c.trim(),
+          ApplicationConstants.CLASS_PATH_SEPARATOR);
     }
-    Apps.addToEnvironment(appMasterEnv, ApplicationConstants.Environment.CLASSPATH.name(),
-        Environment.PWD.$() + File.separator + "*", ApplicationConstants.CLASS_PATH_SEPARATOR);
+    Apps.addToEnvironment(appMasterEnv, classpath, Environment.PWD.$() + File.separator + "*",
+        ApplicationConstants.CLASS_PATH_SEPARATOR);
+    Apps.addToEnvironment(appMasterEnv, classpath, mTachyonHome + "/conf" + File.separator,
+        ApplicationConstants.CLASS_PATH_SEPARATOR);
```
Comment (calvinjia) [inline integration/yarn/src/main/java/tachyon/yarn/Client.java]: Should this be `mTachyonHome + File.separator + "conf" + ...`?

File: integration/yarn/src/main/java/tachyon/yarn/Client.java [hunk 3]
```diff
@@ -297,13 +297,16 @@ private void setupAppMasterJar(LocalResource appMasterJar) throws IOException {
   }
 
   private void setupAppMasterEnv(Map<String, String> appMasterEnv) {
+    String classpath = ApplicationConstants.Environment.CLASSPATH.name();
     for (String c : mYarnConf.getStrings(YarnConfiguration.YARN_APPLICATION_CLASSPATH,
         YarnConfiguration.DEFAULT_YARN_APPLICATION_CLASSPATH)) {
-      Apps.addToEnvironment(appMasterEnv, ApplicationConstants.Environment.CLASSPATH.name(),
-          c.trim(), ApplicationConstants.CLASS_PATH_SEPARATOR);
+      Apps.addToEnvironment(appMasterEnv, classpath, c.trim(),
```
Comment (calvinjia) [inline integration/yarn/src/main/java/tachyon/yarn/Client.java]: `c` is not too descriptive here

File: integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java [hunk 4]
```diff
@@ -81,7 +82,11 @@ public ApplicationMaster(int numWorkers, String tachyonHome, String masterAddres
     mMasterCpu = mTachyonConf.getInt(Constants.MASTER_RESOURCE_CPU);
     mMasterMemInMB = (int) mTachyonConf.getBytes(Constants.MASTER_RESOURCE_MEM) / Constants.MB;
     mWorkerCpu = mTachyonConf.getInt(Constants.WORKER_RESOURCE_CPU);
+    // TODO(binfan): request worker container and ramdisk container separately
+    // memory for running worker
     mWorkerMemInMB = (int) mTachyonConf.getBytes(Constants.WORKER_RESOURCE_MEM) / Constants.MB;
+    // memory for running ramdisk
+    mRamdiskMemInMB =(int) mTachyonConf.getBytes(Constants.WORKER_MEMORY_SIZE) / Constants.MB;
```
Comment (calvinjia) [inline integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java]: `= (int)`

File: integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java [hunk 5]
```diff
@@ -81,7 +82,11 @@ public ApplicationMaster(int numWorkers, String tachyonHome, String masterAddres
     mMasterCpu = mTachyonConf.getInt(Constants.MASTER_RESOURCE_CPU);
     mMasterMemInMB = (int) mTachyonConf.getBytes(Constants.MASTER_RESOURCE_MEM) / Constants.MB;
     mWorkerCpu = mTachyonConf.getInt(Constants.WORKER_RESOURCE_CPU);
+    // TODO(binfan): request worker container and ramdisk container separately
+    // memory for running worker
```
Comment (calvinjia) [inline integration/yarn/src/main/java/tachyon/yarn/ApplicationMaster.java]: Should the changes for this be removed from this PR?

File: integration/bin/tachyon-worker-yarn.sh [hunk 6]
```diff
@@ -13,12 +13,13 @@ mkdir -p "${TACHYON_LOGS_DIR}"
 echo "Formatting Tachyon Worker"
 
 "${JAVA}" -cp "${CLASSPATH}" \
-  ${TACHYON_MASTER_JAVA_OPTS} \
-  -Dtachyon.accesslogger.type="MASTER_ACCESS_LOGGER" \
+  ${TACHYON_WORKER_JAVA_OPTS} \
+  -Dlog4j.configuration=file:$TACHYON_CONF_DIR/log4j.properties \
```
Comment (calvinjia) [inline integration/bin/tachyon-worker-yarn.sh]: no curly braces here?

File: integration/yarn/src/main/java/tachyon/yarn/Client.java (hunk 1 above)
Comment (apc999) [inline integration/yarn/src/main/java/tachyon/yarn/Client.java]: Yes, this should be `static final`, updated.

File: integration/yarn/src/main/java/tachyon/yarn/Client.java (hunk 2 above)
Comment (apc999) [inline integration/yarn/src/main/java/tachyon/yarn/Client.java]: replaced by `PathUtils.concatPath`

File: integration/yarn/src/main/java/tachyon/yarn/Client.java (hunk 3 above)
Comment (apc999) [inline integration/yarn/src/main/java/tachyon/yarn/Client.java]: done

**Comment by calvinjia** [comment]
LGTM

**Comment by yupeng9** [comment]
LGTM

**Pull request merged**